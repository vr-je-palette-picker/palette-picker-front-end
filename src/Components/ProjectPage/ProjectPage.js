import React, { Component } from 'react';
import ProjectContainer from '../ProjectContainer/ProjectContainer.js';
import { Link } from 'react-router-dom';
import { TiArrowBackOutline } from 'react-icons/ti';
import './ProjectPage.scss';

class ProjectPage extends Component {
  createProjectContainers = () => {
    const { projects, fetchProjects } = this.props;
    return projects.map((project, index) => {
      return <ProjectContainer key={index} project={project} fetchProjects={fetchProjects}/>;
    });
  };

  render() {
    return (
    <main>
      <p className='back-button-p'>
        <Link className='link back-button' to='/'>
          <TiArrowBackOutline />
          Back to Palette Picker
        </Link>
        </p>
    <div 
      className='ProjectPage'>{this.createProjectContainers()}
    </div>
    </main>
    )
  }
}

export default ProjectPage;
