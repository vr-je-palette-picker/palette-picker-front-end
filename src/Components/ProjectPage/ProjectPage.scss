.back-button {
  text-decoration: none;
}