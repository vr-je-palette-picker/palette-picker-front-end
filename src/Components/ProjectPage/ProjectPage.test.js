import React from 'react';
import { shallow } from 'enzyme';
import ProjectPage from './ProjectPage';

describe('ProjectPage', () => {
  let wrapper;
  let projects = [{}, {}, {}];

  beforeEach(() => {
    wrapper = shallow(<ProjectPage projects={projects} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
