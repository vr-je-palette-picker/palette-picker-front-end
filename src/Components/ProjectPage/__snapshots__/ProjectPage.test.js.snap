// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPage should match the snapshot 1`] = `
<div
  className="ProjectPage"
>
  <ProjectContainer
    key="0"
    project={Object {}}
  />
  <ProjectContainer
    key="1"
    project={Object {}}
  />
  <ProjectContainer
    key="2"
    project={Object {}}
  />
</div>
`;
